### ISSUE TITLE
Investigate solution for operating the details component with VoiceOver

### ISSUE THREAD

**Issue by owenatgov** [alphagov/govuk-frontend#5310]
## What
Investigate if it's possible to enhance the details component in order for it to be operable via VoiceOver.

## Why
This is in response to https://github.com/alphagov/govuk-design-system/issues/4030. This was raised in a recent DAC audit that screen reader software VoiceOver doesn't announce the details component as interactable. More detail is available in that issue.

## Notes
DAC's solution is similar to that proposed in https://github.com/alphagov/govuk-design-system/issues/4029 which suggests rebuilding the details component to be accordion-esque. We'd like to avoid this if possible as it would [undo work we did to remove the details polyfill for 5.0](https://github.com/alphagov/govuk-frontend/issues/3464).

Ollie previously looked into this issue in https://github.com/alphagov/govuk-frontend/pull/5089 so we should also test this to see if it solves the problem.

## Timebox
1 week (preliminary)

## Questions to answer

- [x] Can we make VoiceOver recognise the details component as a `details` element?
- [x] Does Ollie's solution work?
- [x] Is there a way to reimplement the polyfill progressively?

## Done when

- [x] Questions have been answered or we have a clearer idea of how to get to our goal
- [x] Findings have been reviewed and agreed with at least one other person
- [x] Findings have been shared, e.g: via a write-up on the ticket, at a show & tell or team meeting

**Comment by owenatgov** [comment]
## Update
From some intitial experimentation and discussion, we think that https://github.com/alphagov/govuk-frontend/pull/5089 is the best solution. We _know_ from testing that it solves the problem fro iOS Safari 17.5.1.

Our plan going forward is to:
1. get the above PR ready for review and merge it. We feel it's a useful change, agnostic of this investigation
2. test in older versions of iOS Safari to see if it makes a difference to older versions. This is so that we have these testing records which we can then note in our accessibility strategy

**Comment by selfthinker** [comment]
From testing in older versions of iOS I found that it was worse in iOS 15 and likely the same in iOS 16.
So, this is only fixing the issue in iOS 17 (and surprisingly much older versions like iOS 12).
But that's still a good thing to do as iOS 17 is the predominant iOS version.

More about specific testing results in this comment: https://github.com/alphagov/govuk-frontend/pull/5089#issuecomment-2376966292

**Comment by owenatgov** [comment]
We're closing this investigation and the linked issue as we feel confident that we've solved this. We've noted @selfthinker's fidings above re: the behaviour on older versions of iOS Safari but we're very limited in what we can do here.

**Comment by SteveMilnes** [comment]
Hi @owenatgov. We use the details component in a service I work on in DfE. Did this fix make it into the design system? As in, if our devs use the coded example there, will it work for VoiceOver users?

**Comment by owenatgov** [comment]
Hi @SteveMilnes. I'll respond to both this comment and https://github.com/alphagov/govuk-frontend/issues/5308#issuecomment-3136183613 here:

In summary, this issue with voiceover was addressed. The issue with dragon is still present and, until dragon resolve this at the vendor level, there's very little we can do. There's a summary of our discussion with DAC in [this comment](https://github.com/alphagov/govuk-design-system/issues/4194#issuecomment-2496246948) where DAC recommended we don't proceed with our solution to add `aria-controls` to the details element as it causes problems in numerous unpredictable ways.
